src/Microsoft.ML.Transforms/CategoricalTransform.cs

```diff
@@ -301,4 +312,116 @@ public static CommonOutputs.TransformOutput KeyToText(IHostEnvironment env, KeyT
+
+            public Config(OneHotOutputKind outputKind, KeyValueOrder order, int max)
+            {
+                OutputKind = outputKind;
+                Order = order;
+                Max = max;
+            }
+        }
+
+        private interface ICategoricalCol
+        {
+            PipelineColumn Input { get; }
+            Config Config { get; }
+        }
+
+        private sealed class ImplScalar<T> : Vector<float>, ICategoricalCol
+        {
+            public PipelineColumn Input { get; }
+            public Config Config { get; }
+            public ImplScalar(PipelineColumn input, Config config) : base(Rec.Inst, input)
+            {
+                Input = input;
+                Config = config;
+            }
+        }
+
+        private sealed class ImplVector<T> : Vector<float>, ICategoricalCol
+        {
+            public PipelineColumn Input { get; }
+            public Config Config { get; }
+            public ImplVector(PipelineColumn input, Config config) : base(Rec.Inst, input)
+            {
+                Input = input;
+                Config = config;
+            }
+        }
+
+        private sealed class Rec : EstimatorReconciler
+        {
+            public static readonly Rec Inst = new Rec();
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env, PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames, IReadOnlyDictionary<PipelineColumn, string> outputNames, IReadOnlyCollection<string> usedNames)
+            {
+                var infos = new CategoricalEstimator.ColumnInfo[toOutput.Length];
+                for (int i = 0; i < toOutput.Length; ++i)
+                {
+                    var tcol = (ICategoricalCol)toOutput[i];
+                    infos[i] = new CategoricalEstimator.ColumnInfo(inputNames[tcol.Input], outputNames[toOutput[i]], (CategoricalTransform.OutputKind)tcol.Config.OutputKind,
+                        tcol.Config.Max, (TermTransform.SortOrder)tcol.Config.Order);
+                }
+                return new CategoricalEstimator(env, infos);
+            }
+        }
+
+        /// <summary>
+        /// Converts the categorical value into an indicator array by building a dictionary of categories based on the data and using the id in the dictionary as the index in the array
+        /// </summary>
+        /// <param name="input">Incoming data.</param>
+        /// <param name="outputKind">Specify output type of indicator array: Multiarray, array or binary encoded data.</param>
+        /// <param name="order">How Id for each value would be assigined: by occurence or by value.</param>
+        /// <param name="maxItems">Maximum number of ids to keep during data scanning.</param>
+        public static Vector<float> OneHotEncoding(this Scalar<string> input, OneHotOutputKind outputKind = DefOut, KeyValueOrder order = DefSort, int maxItems = DefMax)
```

Comment:
>OneHotEncoding [](start = 36, length = 14)

It strikes me that the most useful thing that we could get out of this via an `onFit` delegate would be the mapping, so people can know *what* the features actually mean.

Reply:
This could be the same type that is from the `OnTerm` transform... then once that is properly implemented both with benefit.

---
In reply to: [217569940](https://github.com/dotnet/machinelearning/pull/899#discussion_r217569940) [](ancestors = 217569940)

Reply:
Good thanks Ivan.

---
In reply to: [218176803](https://github.com/dotnet/machinelearning/pull/899#discussion_r218176803) [](ancestors = 218176803,217570555,217569940)